portality/migrate/20220928_1735_missing_admin/operations.py

```diff
@@ -0,0 +1,24 @@
+from portality.models import Journal
+def add_admin_field(article):
+    log = ""
+    if not "admin" in article.data:
+        print(article.data["index"]["issn"])
+        try:
+            j_data = Journal.find_by_issn(article.data["index"]["issn"])
+            if len(j_data):
+                log += "Article matches more than one journal: "
+                log += j_data
+            else:
+                j = Journal.pull(j_data[0]["id"])
+        except:
+            log += "\n Corrupted article, no issn: ";
+            log += article.data;
+        if j is not None:
+            article.data["admin"] = {"in_doaj": j.is_in_doaj}
+        else:
+            log += "\n Article with no journal found: ";
+            log += article.data;
+    if log != "":
+        with open('1735_log.txt','w') as file:
+            file.write(log);
```

Comment:
Isn't this going to overwrite the file every time, so you will only have the log for the last article run?  If you want to append to a file, you need to open it with mode `a`. 

The upgrade framework just uses `print` statements, so you could do that instead, and then when you run the script you can direct the output into a file, that way you'll have all the logs.

Reply:
I'll change it to `prints`, thanks!